File: lab-5/src/com/company/commands/AddCmd.java
package com.company.commands;

import com.company.collection.MusicBand;
import com.company.collection.MusicBandStorage;
import com.company.container.Message;
import com.company.util.IOStream;

public class AddCmd extends Command {
   public AddCmd(MusicBandStorage target, String cmdName, IOStream ioStream) {
      super(target, cmdName, ioStream,
              "Добавить элемент в коллекцию"
      );
   }

   @Override
   public Message exec(Message message){
      //TODO: do some shit
      ioStream.writelnterm("ДОБАВЛЕНИЕ НОВОГО ЭЛЕМЕНТА В КОЛЛЕКЦИЮ " + target.getName());
      ioStream.writelnterm("Введите значения следующих свойств: ");
      MusicBand musicBand = ioStream.readMusicBand();
      target.add(musicBand);
      ioStream.writelnterm("(Новый элемент добавлен)");


      return message;
   }

}

File: lab-6-kotlin/src/main/kotlin/com/github/e1turin/lab5/commands/AddCmd.kt
package com.github.e1turin.lab5.commands

import com.github.e1turin.lab5.collection.MusicBand
import com.github.e1turin.lab5.collection.MusicBandStorage
import com.github.e1turin.lab5.containers.Message
import com.github.e1turin.lab5.containers.Response
import com.github.e1turin.lab5.containers.ResponseType
import com.github.e1turin.lab5.util.IOStream

class AddCmd(cmdName: String) :
    Command(cmdName, "Добавить новый элемент в коллекцию") {
    override fun exec(arg: String, target: MusicBandStorage, ioStream: IOStream): Message {
        ioStream.writeln("ДОБАВЛЕНИЕ НОВОГО ЭЛЕМЕНТА В КОЛЛЕКЦИЮ " + target.name)
        val musicBand: MusicBand = ioStream.readMusicBand(
            "Необходимо заполнить значения следующих свойств: "
        )
        musicBand.setId(target.nextElementId())
        target.add(musicBand)
        ioStream.writeln("Новый элемент добавлен")

        return Response(
            cmdName, ResponseType.TASK_COMPLETED,
            content = "$cmdName executed with argument: arg='$arg'"
        )
    }
}
